composables/transaction/mintCollection/useNewCollectionId.ts

```diff
@@ -1,32 +1,20 @@
-import { getRandomValues } from '@/components/unique/utils'
-import { mapToId } from '@/utils/mappers'
-import { unwrapSafe } from '@/utils/uniquery'
-
 export function useNewCollectionId() {
+  const { apiInstance } = useApi()
   const newCollectionId = ref<number>()
-  const randomNumbers = getRandomValues(10).map(String)
 
-  const { data, error, loading, refetch } = useGraphql({
-    queryName: 'existingCollectionList',
-    variables: {
-      ids: randomNumbers,
-    },
-  })
+  const getCollectionId = async () => {
+    const api = await apiInstance.value
+    const nextCollectionId = (
+      await api.query.nfts.nextCollectionId()
+    ).value.toNumber()
+    newCollectionId.value = nextCollectionId
+  }
 
-  watch(data, () => {
-    if (data.value) {
-      const collectionEntities = data.value.collectionEntities
-      const collectionList = unwrapSafe(collectionEntities)
-      const existingIds = collectionList.map(mapToId)
-      const newId = randomNumbers.find((id) => !existingIds.includes(id))
-      newCollectionId.value = Number(newId)
-    }
+  getCollectionId().catch((e) => {
+    console.error('Error:', e)
```

Comment:
this should be done via subscribe

https://polkadot.js.org/docs/api/start/api.query.subs